jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbDiagnosticsCollector.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+* Copyright (c) 2020, 2022 IBM Corporation, Matheus Cruz and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation, Matheus Cruz - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4jakarta.jdt.core.jsonb;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.eclipse.jdt.core.Flags;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.IMember;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.ISourceRange;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4jakarta.jdt.core.DiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.JDTUtils;
+import org.eclipse.lsp4jakarta.jdt.core.JakartaCorePlugin;
+
+/**
+ * This class contains logic for Jsonb diagnostics:
+ * 1) Multiple JsonbCreator annotations on constructors will cause a diagnostic.
+ * 2) JsonbTransient not being a mutually exclusive Jsonb annotation will cause a diagnostic. 
+ */
+public class JsonbDiagnosticsCollector implements DiagnosticsCollector {
+    
+    private final String JSONB_CREATOR_DIAGNOSTIC_TYPE = "JsonbCreator";
+    private final String JSONB_TRANSIENT_DIAGNOSTIC_TYPE = "JsonbTransient";
```

Comment:
They are already defined in the JsonbConstants class. We can delete these here and use the ones in JsonbConstants instead.

Reply:
Done